OverlayPlugin.Core/MemoryProcessors/Combatant/LineCombatant.cs

```diff
@@ -0,0 +1,411 @@
+                            combatant.ID,
+                            string.Join("", changed.Select((fi) => FormatFieldChange(fi, combatant))));
+                        continue;
+                    }
+                }
+
+                // Check for partial change lines with a delay less than the default delay
+                // Batch these partial changes
+                var delayNameID = inCombat ? CombatantChangeCriteria.InCombatDelayNameID : CombatantChangeCriteria.OutOfCombatDelayNameID;
+                if (lastUpdatedDiff > delayNameID && combatant.BNpcNameID != oldCombatant.BNpcNameID)
+                {
+                    changed.Add(combatant.GetType().GetField("BNpcNameID"));
+                }
+                var delayTransformationID = inCombat ? CombatantChangeCriteria.InCombatDelayTransformationID : CombatantChangeCriteria.OutOfCombatDelayTransformationID;
+                if (lastUpdatedDiff > delayTransformationID && combatant.TransformationId != oldCombatant.TransformationId)
+                {
+                    changed.Add(combatant.GetType().GetField("TransformationId"));
+                }
+                var delayWeaponID = inCombat ? CombatantChangeCriteria.InCombatDelayWeaponID : CombatantChangeCriteria.OutOfCombatDelayWeaponID;
+                if (lastUpdatedDiff > delayWeaponID && combatant.WeaponId != oldCombatant.WeaponId)
+                {
+                    changed.Add(combatant.GetType().GetField("WeaponId"));
+                }
+                var delayTargetID = inCombat ? CombatantChangeCriteria.InCombatDelayTargetID : CombatantChangeCriteria.OutOfCombatDelayTargetID;
+                if (lastUpdatedDiff > delayTargetID && combatant.TargetID != oldCombatant.TargetID)
+                {
+                    changed.Add(combatant.GetType().GetField("TargetID"));
+                }
+                var delayPosition = inCombat ? CombatantChangeCriteria.InCombatDelayPosition : CombatantChangeCriteria.OutOfCombatDelayPosition;
+                if (lastUpdatedDiff > delayPosition)
+                {
+                    // This check seems redundant but it's less expensive than the check below against distance
+                    // so it uses less CPU
+                    if (combatant.PosX != oldCombatant.PosX || combatant.PosY != oldCombatant.PosY || combatant.PosZ != oldCombatant.PosZ)
+                    {
+                        var dist = Math.Sqrt(
+                            Math.Pow(combatant.PosX - oldCombatant.PosX, 2)
+                            + Math.Pow(combatant.PosY - oldCombatant.PosY, 2)
+                            + Math.Pow(combatant.PosZ - oldCombatant.PosZ, 2));
+                        var checkDist = inCombat ? CombatantChangeCriteria.InCombatDistancePosition : CombatantChangeCriteria.OutOfCombatDistancePosition;
+                        if (dist > checkDist)
+                        {
+                            if (combatant.PosX != oldCombatant.PosX)
```

Comment:
I'm starting to write some code on my end to support this, and I realize that at least for me, it's a lot easier if x/y/z/heading are treated as one unit (i.e. write them all even if only one changed). May be easier for other use cases too, where you want to check if a combatant position is within a particular area based on the log line.

Reply:
(Mostly my own curiosity here, but what sorts of programming structures does this make easier if you're doing something like https://github.com/OverlayPlugin/OverlayPlugin/pull/195#discussion_r1137733727 that you mentioned before?)

Reply:
Mostly just that I treat a position as one single record. It doesn't *break* anything if I have to do it differently, just adds more logic.

Reply:
Does this end up adding a bunch to the log file size? Or is it often the case that these are changed together? Mostly just wondering about the trade-off of "is a more efficient log file better than a little bit of code in downstream log processors".

Reply:
I realize actually the main use case would probably be individual line-based triggers, rather than anything on my end. It would remove some need for keeping state, since you could just wait for a line with the desired x/y/z/heading all in one go. 

I think the more likely issue is that since x/y (and to a lesser degree, heading) would *usually* show up at the same time when something moves, someone looking at the log lines might assume that they can rely on those all being present, when that's not the case. Since they'd all typically change together (with the exception of Z which is likely to be mostly noise from players jumping and such), I don't think it would bloat a log *that much* to always include all four. Looking at the p8s2 sample log, there are very few "Change" lines where X changes but not Y or vice-versa, though Z and heading are a different story.

Reply:
Position and Heading will need special case handling regardless because they're written to check thresholds. Removing the "always write all four" logic actually increases the line count (and the emitted IL code/CPU usage) because it'll need to check for each individual value of PosX, PosY, PosZ has changed separately to determine which need added to the line.